tests/test_url_handler.py

```diff
@@ -0,0 +1,696 @@
+    form_data = {
+        'nominator_email': 'somebody@somewhere.com',
+        'nominator_name': 'John Smith',
+        'nominator_institution': 'UNT'
+    }
+
+    assert len(models.Nominator.objects.all()) == 1
+    new_nominator = url_handler.get_nominator(form_data)
+    assert len(models.Nominator.objects.all()) == 2
+    for key in form_data.keys():
+        assert getattr(new_nominator, key) == form_data[key]
+
+
+def test_get_nominator_when_nominator_cannot_be_created():
+    form_data = {
+        'nominator_email': 'somebody@somewhere.com',
+        'nominator_name': None,
+        'nominator_institution': None
+    }
+    new_nominator = url_handler.get_nominator(form_data)
+
+    assert new_nominator is False
+
+
+@pytest.mark.parametrize('scope_value, expected', [
+    ('1', 'You have already declared {} as "In Scope"'),
+    ('0', 'You have already declared {} as "Out of Scope"')
+])
+def test_nominate_url_nomination_exists(scope_value, expected):
+    project = factories.ProjectFactory()
+    nominator = factories.NominatorFactory()
+    form_data = {'url_value': 'http://www.example.com'}
+    factories.NominatedURLFactory(
+        url_nominator=nominator,
+        url_project=project,
+        entity=form_data['url_value'],
+        value=scope_value
+    )
+
+    assert (url_handler.nominate_url(project, nominator, form_data, scope_value) ==
+            [expected.format(form_data['url_value'])])
+
+
+@pytest.mark.parametrize('scope_value, expected', [
+    ('1', 'You have successfully declared {} as "In Scope"'),
+    ('0', 'You have successfully declared {} as "Out of Scope"')
+])
+def test_nominate_url_nomination_modified(scope_value, expected):
+    project = factories.ProjectFactory()
+    nominator = factories.NominatorFactory()
+    form_data = {'url_value': 'http://www.example.com'}
+    factories.NominatedURLFactory(
+        url_nominator=nominator,
+        url_project=project,
+        entity=form_data['url_value'],
+        value='1' if scope_value == '0' else '0'
+    )
+
+    assert (url_handler.nominate_url(project, nominator, form_data, scope_value) ==
+            [expected.format(form_data['url_value'])])
+
+
+def test_nominate_url_new_nomination():
+    project = factories.ProjectFactory()
+    nominator = factories.NominatorFactory()
+    form_data = {'url_value': 'http://www.example.com'}
+    scope_value = 1
+    expected = ['You have successfully nominated {0}'.format(form_data['url_value'])]
+
+    assert url_handler.nominate_url(project, nominator, form_data, scope_value) == expected
+
+
+def test_nominate_url_cannot_create_nomination():
+    project = None
+    nominator = None
+    form_data = {}
+    scope_value = None
+
+    with pytest.raises(http.Http404):
+        url_handler.nominate_url(project, nominator, form_data, scope_value)
+
+
+@pytest.mark.parametrize('attr_value', [
+    ['some_value', ],
+    ['some_value', 'some_other_value']
+])
+def test_add_other_attribute(attr_value):
```

Comment:
Yeah I'll take a look at this and see how best to simplify it. The issue right now is that the output is based upon metadata objects that have randomized values and valuesets. But I'll figure something out.